lib/slop/result.rb

```diff
@@ -14,12 +14,23 @@ def initialize(parser)
       @options = parser.options
     end
 
-    # Returns an options value, nil if the option does not exist.
+    # Returns an option's value, nil if the option does not exist.
     def [](flag)
       (o = option(flag)) && o.value
     end
     alias get []
 
+    # Returns an option's value, raises KeyError if the option does not exist.
+    def fetch(flag)
+      o = option(flag)
+      if o.nil?
+        cleaned_key = clean_key(flag)
+        raise UnknownOption.new("key not found: '#{cleaned_key}'", "#{cleaned_key}")
```

Comment:
One last nitpick I promise :) 

I think the error message should have more detail. "key not found" might be confusing in the context of command line options and flags. After all, what is a key? Perhaps we could just re-use `option` like we did for the other `UnknownOption` class ("option not found") to clear this up? That might be enough to avoid confusion. What do you think?

Reply:
Don't worry, I also really want to have this feature as good as possible so don't worry!
If you have any other suggestion I'm open to them :-)

Thanks for your feedback!